core/domain/state_domain.py

```diff
@@ -1490,12 +1498,30 @@ def validate(self):
 
         if not (self.data_format == self.DATA_FORMAT_UNICODE or
                 self.data_format == self.DATA_FORMAT_HTML):
-            raise utils.ValidationError(
-                'Invalid data_format: %s' % self.data_format)
+            try:
+                obj = getattr(objects, self.data_format)
+            except AttributeError:
+                raise utils.ValidationError(
+                    'Invalid data_format: %s' % self.data_format)
 
-        if not isinstance(self.translation, python_utils.BASESTRING):
-            raise utils.ValidationError(
-                'Invalid translation: %s' % self.translation)
+            if not issubclass(obj, objects.BaseTranslatableObject):
+                raise utils.ValidationError(
+                    'Invalid data_format: %s' % self.data_format)
+
+        if (
+                self.data_format == self.DATA_FORMAT_UNICODE or
+                self.data_format == self.DATA_FORMAT_HTML
+        ):
+            if not isinstance(self.translation, python_utils.BASESTRING):
+                raise utils.ValidationError(
+                    'Invalid translation: %s' % self.translation)
+        else:
+            try:
+                obj = getattr(objects, self.data_format)
+                obj.normalize_value(self.translation)
+            except TypeError:
+                raise utils.ValidationError(
+                    'Invalid translation: %s' % self.translation)
```

Comment:
Maybe move these validation errors into a separate `validate` function and call that here?